File: backend/src/main/java/com/iapp/iapp_messenger/dao/hibernate/PaymentType.java
package com.iapp.iapp_messenger.dao.hibernate;

/// Допустимые варианты оплаты; в сущности Family значение хранится строкой.
public enum PaymentType {
    CASH("Наличные"),
    TRANSFER("Перевод"),
    QR_CODE("QR-код"),
    DEBT("Долг");

    private final String title;

    PaymentType(String title) {
        this.title = title;
    }

    public String getTitle() {
        return title;
    }
}

File: frontend/src/main/kotlin/com/interactme/data/PaymentType.kt
package com.interactme.data

/// Варианты оплаты в интерфейсе; в DTO и JSON сохраняется строковое значение.
enum class PaymentType(val title: String)
{
    Cash("Наличные"),
    Transfer("Перевод"),
    QrCode("QR-код"),
    Debt("Долг");

    companion object {
        /// Находит enum по строке, пришедшей из backend или выбранной в интерфейсе.
        fun fromTitle(title: String?): PaymentType? = entries.firstOrNull { it.title == title }
    }
}